fabric-events-interaction-v0/src/main/java/net/fabricmc/fabric/api/event/player/ItemPickupEvent.java

```diff
@@ -0,0 +1,29 @@
+package net.fabricmc.fabric.api.event.player;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.ActionResult;
+import net.minecraft.util.Hand;
+import net.minecraft.util.TypedActionResult;
+import net.minecraft.world.World;
+
+public interface ItemPickupEvent {
+	Event<ItemPickupEvent> EVENT = EventFactory.createArrayBacked(ItemPickupEvent.class,
+			listeners -> (player, itemStack) -> {
```

Comment:
I believe the question that held up past attempts was whether entities other than players would handle this

Reply:
If there isn't a consistent, easy way to detect this then there's too much complexity involved, since what are you going to do with modded entities?

Reply:
@TheGlitch76 
Modded entities can fire this event themselves, if its modified to not be player-specific.